pages/docs/reference/compiler-reference.md

```diff
@@ -0,0 +1,354 @@
+
+### `-progressive`
+* Enable the [progressive mode](whatsnew13.html#progressive-mode) for the compiler.
+    
+    In the progressive mode, deprecations and bug fixes for unstable code take effect immediately,
+    instead of going through a graceful migration cycle.
+    Code written in the progressive mode is backward compatible; however, code written in
+    non-progressive mode may cause compilation errors in the progressive mode.
+    
+{:#argfile}
+
+### `@<argfile>`
+* Read the compiler options from the given file. A file passed in this option can contain compiler options with values 
+and paths to source files. For example:
+
+    <div class="sample" markdown="1" mode="shell" theme="idea">
+    
+    ```
+    -d hello.jar -include-runtime
+    hello.kt
+    ```
+    
+    </div>
+
+    You can also separate compiler options from source file names and pass multiple argument files.
+
+    <div class="sample" markdown="1" mode="shell" theme="idea">
+    
+    ```bash
+    $ kotlinc @compiler.options @classes
+    ```
+    
+    </div>
+    
+    If the files reside in locations different from the current directory, use relative paths. 
+    
+    <div class="sample" markdown="1" mode="shell" theme="idea">
+    
+    ```bash
+    $ kotlinc @options\compiler.options hello.kt
+    ```
+    
+    </div>
+
+    
+## JVM compiler options
+
+Kotlin compiler for JVM compiles Kotlin source files into Java class files. 
+The command-line tools for Kotlin to JVM compilation are `kotlinc` and `kotlinc-jvm`.
+You can also use them for executing Kotlin script files.
+
+In addition to [common options](#common-options), Kotlin/JVM compiler has the options listed below.
+
+### `-classpath <path>` (`-cp <path>`)
+* Search for class files in the specified paths. Separate elements of the classpath with semicolons (**;**).
+The classpath can contain file and directory paths, ZIP, or JAR files.
+
+### `-d <path>`
+* Place the generated class files into the specified location. The location can be a directory, a ZIP, or a JAR file. 
+
+### `-include-runtime`
+* Include the Kotlin runtime into the resulting JAR file. Makes the resulting archive runnable on any Java-enabled 
+environment.
+
+### `-jdk-home <path>`
+* Use a custom JDK home directory to include into classpath if it differs from the default `JAVA_HOME`.
+
+### `-jvm-target <version>`
+* Specify the target version of the generated JVM bytecode. Possible values are `1.6`, `1.8`, `9`, `10`, `11`, and `12`.
+ The default value is `1.6`.
+
+### `-java-parameters`
+* Generate metadata for Java 1.8 reflection on method parameters.
+
+### `-module-name <name>`
+* Set the custom name for the generated `.kotlin_module` file.
+  
+### `-no-jdk`
+* Exclude the Java runtime from the classpath.
+
+### `-no-reflect`
+* Exclude `kotlin-reflect.jar` from the classpath.
+  
+### `-no-stdlib`
+* Exclude both `kotlin-stdlib.jar` and `kotlin-reflect.jar` from the classpath. 
+  
+### `-script <file>`
```

Comment:
In fact, the `-script` argument is boolean, and it makes the compiler run the first script from the set of compiled files. (E.g. `kotlinc 1.kt 2.kts 3.kt -script` is valid, and will compile all files and run `2.kts`.)

Also note that since 1.3.70, this will be a common argument (https://github.com/JetBrains/kotlin/commit/f59e393e37d1d745261e6a2db9921b3fe5ccf7fc)